src/customScreens.ts

```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import type { MediaScreen, PlaylistScreen, ScreenMap } from '#types/screens';
+
+/**
+ * Content type to screen mapping
+ *
+ * After a media or playlist is fetched, the router will try to match a custom screen based on the `contentType`
+ * property. If no match is found, it will fall back to the `default` screen.
+ *
+ * This is useful for two reasons:
+ *
+ * 1) You want to use a different movie screen
+ * 2) You want to support additional screens based on the metadata. Examples are; hub, vlog, collections, etc...
+ *
+ * The screens are imported used `React.lazy` to enable lazy loading the screens when requested.
+ */
+
+// lazy import media screens
+const mediaHub = React.lazy(() => import('./screens/MediaHub/MediaHub'));
```

Comment:
This bit is going to be prone to merge conflicts if anyone tries to develop custom screens.  What if we rename the base folder here to 'pages' and move all the 'custom' screens into "pages/media" or "pages/screens" or "pages/mediaScreens" or something?  Then hopefully we can programmatically iterate over these and reduce the boilerplate.

Reply:
The more I look at this the more it feels like differentiating a Page (web app level route) from a Screen makes sense. We can have pages/screens/media and pages/screens/playlist directories.  Then we just need a convention for how to get the content type (maybe we require the default export for each of these screens to be {contentType, screenComponent})

Reply:
I get what you mean. So to agree on the terminology, pages are the routes defined by the application, and screens are rendered within a page (custom or default screen).

The problem with an export like `{contentType, screenComponent}` is that we can't lazy load the component anymore because we need that contentType to determine which component to use. Also, there is a chance the same screen is used for multiple contentTypes.

Reply:
OK, programmatic loading would be cool, but maybe it's a bridge too far for now. Just to sanity check, what exactly is the lazy loading buying us? Would there be that much overhead to load the screens normally?